src/tiled/automappingmanager.cpp

```diff
@@ -205,21 +206,53 @@ bool AutomappingManager::loadFile(const QString &filePath)
 void AutomappingManager::setMapDocument(MapDocument *mapDocument)
 {
     cleanUp();
-    if (mMapDocument)
+
+    QString oldRules;
+    if (mMapDocument) {
         mMapDocument->disconnect(this);
+        oldRules = getRulesFileName();
+        QFileInfo oldFile(oldRules);
+        if (oldFile.exists()) {
+            mWatcher->disconnect(SIGNAL(fileChanged(const QString &)));
+            mWatcher->removePath(oldRules);
+        }
+    }
 
     mMapDocument = mapDocument;
 
+    QString newRules;
     if (mMapDocument) {
         connect(mMapDocument, &MapDocument::regionEdited,
                 this, &AutomappingManager::onRegionEdited);
+        newRules = getRulesFileName();
+        QFileInfo newFile(newRules);
+        if (newFile.exists()) {
+            mWatcher->addPath(newRules);
+            connect(mWatcher, SIGNAL(fileChanged(const QString &)),
+                this, SLOT(onFileChanged()));
```

Comment:
We don't need to connect/disconnect to this signal all the time. It can be made once in the constructor instead, and the disconnect will happen automatically.

Also, please use the new connect style:
```c++
    connect(mWatcher, &QFileSystemWatcher::fileChanged,
            this, &AutomappingManager::onFileChanged);
```